### ISSUE TITLE
Headers not propagating back to caller via zuul?

### ISSUE THREAD

**Issue by arun0009** [spring-cloud/spring-cloud-netflix#719]
Looks like the headers are not propagating back to caller via zuul? I had to get headers from zuul response headers and set it to http response headers?

```
    private void setResponse(RequestContext context, Response response) throws IOException {
        this.helper.setResponse(response.code(), response.body() == null ? null : response.body().byteStream(), getHeaders(response.headers()));
        for(Pair<String, String> header: context.getZuulResponseHeaders()){
            context.getResponse().setHeader(header.first(), header.second());
        }
    }
```

**Comment by spencergibb** [comment]
Can you explain in a little more detail what's going on? Where did you put that code?

**Comment by arun0009** [comment]
We wrote our own Custom Filter and disabled Spring Cloud Zuul provided filters by adding:

<code>
zuul:
  RibbonRoutingFilter:
    route:
      disable: true
  SimpleHostRoutingFilter:
      route:
        disable: true
</code>

Client --> Zuul --> Service

In this scenario Service returns a few headers that I have to pass back to client, however Zuul ignores those headers and just returns <b>X-Application-Context</b>: zuulproxy:local:8080 to Client. So we had to update the setResponse method to pass headers back to client. Shouldn't the helper method of ProxyRequestHelper.setResponse set the Client response headers?
